src/utils.ts

```diff
@@ -48,3 +48,40 @@ export async function captureError(fn: () => Promise<any>): Promise<string> {
 export function isTopLevelAccount(accountId: string): boolean {
   return accountId.includes('.');
 }
+
+export function urlConfigFromNetwork(network: string | {network: string; port?: number}): ClientConfig {
+  const networkName = typeof network === 'string' ? network : network.network;
+  switch (networkName) {
+    case 'sandbox':
+      if (typeof network === 'string') {
+        throw new TypeError('Sandbox\'s network argument can\'t be a string');
+      }
+
+      if (network.port === undefined) {
+        throw new TypeError('Sandbox\'s network.port is not defined');
+      }
+        return {
+          network: 'sandbox',
+          rpcAddr: `http://localhost:${network.port!}`,
```

Comment:
I find it odd that `sandbox` gets such special treatment. Why does it need it? Why not just export `rpcAddr: 'http://localhost'` and add the port at the point when it is needed?

Also, if we get rid of `JsonRpcProvider.fromNetwork` and `JsonRpcProvider.archival` for now, I don't think `urlConfigFromNetwork` will be used anywhere. Is that correct? Should `runtime.ts` use this in its `clientConfig` methods?

Reply:
The thought with this function is that currently these fields are set in the corresponding `Runtime`, but instead they should go through this function and the sandbox will need to pass the port.  But perhaps you are right and we can just provide the string as is and it's the sandbox runtime's job to add the port.